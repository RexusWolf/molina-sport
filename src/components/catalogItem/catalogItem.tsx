import React from 'react';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

interface CardProps {
  productName: string
  productImg: string
  productDescription: string
  productPrice: number
}

export const CatalogItem: React.FC<CardProps> = ({ productName, productImg, productDescription, productPrice }) => (
  <Card>
    <CardActionArea>
      <CardMedia
        component="img"
        height="320"
        image={productImg}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2" data-testid="item-name">
          {productName}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p" data-testid="item-description">
          {productDescription}
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Grid container spacing={3}>
        <Grid item xs>
          <Button color="primary">
            Ver producto
            </Button>
        </Grid>
        <Grid item xs>
          <Typography variant="h4" data-testid="item-price">{productPrice}â‚¬</Typography>
        </Grid>
      </Grid>
    </CardActions>
  </Card >
);