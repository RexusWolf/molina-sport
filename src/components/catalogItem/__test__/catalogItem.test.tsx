import React from 'react';
import ReactDOM from 'react-dom'
import { CatalogItem } from '../catalogItem'

import { render, cleanup } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CatalogItem productName="name" productDescription="description" productImg="imgurl" productPrice={22} ></CatalogItem >, div);
  ReactDOM.unmountComponentAtNode(div);
})

it("renders item correctly", () => {
  const { getByTestId } = render(<CatalogItem productName="name" productDescription="description" productImg="imgurl" productPrice={22} ></CatalogItem >);
  expect(getByTestId('item-name')).toHaveTextContent("name")
  expect(getByTestId('item-description')).toHaveTextContent("description")
  expect(getByTestId('item-price')).toHaveTextContent("22â‚¬")
})

it("matches snapshot", () => {
  const snap = renderer.create(<CatalogItem productName="name" productDescription="description" productImg="imgurl" productPrice={22} ></CatalogItem >).toJSON();
  expect(snap).toMatchSnapshot();
})