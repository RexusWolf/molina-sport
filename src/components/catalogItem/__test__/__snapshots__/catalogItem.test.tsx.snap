// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "height": "520px",
    }
  }
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      data-testid="item-img"
      height="300px"
      src="imgurl"
    />
    <div
      className="MuiCardContent-root"
      style={
        Object {
          "height": "120px",
        }
      }
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        data-testid="item-name"
      >
        name
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        data-testid="item-description"
      >
        description
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Ver producto
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
          data-testid="item-price"
        >
          22
          â‚¬
        </h4>
      </div>
    </div>
  </div>
</div>
`;
